src/main/java/mate/academy/bookshop/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.bookshop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.user.UserRegistrationRequestDto;
+import mate.academy.bookshop.dto.user.UserResponseDto;
+import mate.academy.bookshop.exception.RegistrationException;
+import mate.academy.bookshop.mapper.UserMapper;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.repository.UserRepository;
+import mate.academy.bookshop.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto save(UserRegistrationRequestDto dto) {
+        if (existsByEmail(dto.getEmail())) {
+            throw new RegistrationException("This email is already used");
+        }
+        User user = userRepository.save(userMapper.toModel(dto));
+        return userMapper.toDto(user);
+    }
+
+    @Override
+    public boolean existsByEmail(String email) {
+        return userRepository.existsByEmail(email);
+    }

```

Comment:
not sure it makes too much sense to extract such a small part of code, better to extract the whole part with `if (exists) { throw new Ex();}` and name it `checkEmail(email)`